brontide/read_buffer_pool_test.go

```diff
@@ -0,0 +1,57 @@
+package brontide_test
+
+import (
+	"testing"
+	"time"
+
+	"github.com/lightningnetwork/lnd/brontide"
+)
+
+// TestReadBufferPool verifies that the buffer pool properly resets used read
+// buffers.
+func TestReadBufferPool(t *testing.T) {
```

Comment:
Should be possible to abstract that other test a bit in order to make it a table driven tests of the `queue.Recycler` implementations.

Reply:
this can get become involved with the current layout since they live in different packages.

there could be an argument for structuring the packages like:
```
buffer/
  read.go
  write.go
pool/
  read_buffer.go
  write_buffer.go
```
is that something we want to do here or in a followup?

Reply:
IMO can be done in #2474.